luma/core/framebuffer.py

```diff
@@ -89,33 +96,18 @@ class full_frame(object):
     :param device: The target device, used to determine the bounding box.
     :type device: luma.core.device.device
     """
-    def __init__(self, device):
-        self.bounding_box = (0, 0, device.width, device.height)
 
-    def redraw_required(self, image):
+    def __init__(self, **kwargs):
```

Comment:
is this necessary to override?

Reply:
I think the implementations will call this dynamically passing the device like here: https://github.com/rm-hull/luma.oled/blob/master/luma/oled/device/color.py#L25, and I may want to pass in the num_segments too, so we can either use `**kwargs` or have `device, num_segments` like the diff-to-previous constructor

Reply:
in that case, maybe add a docstring so you can omit the `pass` statement altogether.

Reply:
Refactored API so constructor isn't needed at all